### PR TITLE
suite fixes

### PR DESCRIPTION
**Pull request by yromanyshyn**

### CODE REVIEW

**Review by noamzbr — APPROVED** [review]
Looks good, but maybe we should make it so that trying to access context.train if not set will return None rather than raise an exception?

**Comment by yromanyshyn** [comment]
@noamzbr 

> Looks good, but maybe we should make it so that trying to access context.train if not set will return None rather than raise an exception?

does not this contradict the initial purpose of the context/
@matanper

**Comment by matanper** [comment]
> @noamzbr
> 
> > Looks good, but maybe we should make it so that trying to access context.train if not set will return None rather than raise an exception?
> 
> does not this contradict the initial purpose of the context/ @matanper

Right because then in every check we will have to test it instead of just in context

File: deepchecks/vision/base.py [hunk 1]
```diff
@@ -362,102 +363,160 @@ def run(
...
+                results=results
+            )
+
+        for check_idx, check in checks.items():
+            if check_idx not in results:
+                try:
+                    if isinstance(check, SingleDatasetCheck):
+                        results[check_idx] = check.compute(
+                            context,
+                            dataset_kind=None # TODO: when this method should be called?
```
Comment (matanper) [inline deepchecks/vision/base.py:420]: This should be the appropriate dataset

Comment (yromanyshyn) [reply deepchecks/vision/base.py:420]: I understand this) but this code is outside training/validation loops, which dataset is appropriate?

Comment (yromanyshyn) [reply deepchecks/vision/base.py:420]: should it be called twice if train and test were passed?

Comment (yromanyshyn) [reply deepchecks/vision/base.py]: fixed, resolving

**Pull request merged**